### PR TITLE
AWS S3: Allow source keys with non-ASCII characters

### PR DESCRIPTION
**Pull request by mpdn**
Adds missing URL encoding for source keys when copying.

### CODE REVIEW

File: s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala [hunk 1]
```diff
@@ -508,6 +475,47 @@ trait S3IntegrationSpec
       .runWith(Sink.head)
       .futureValue shouldEqual akka.Done
   }
+
+  private def uploadCopyDownload(sourceKey: String, targetKey: String): Assertion = {
+    val source: Source[ByteString, Any] = Source(ByteString(objectValue) :: Nil)
```
Comment (ennru) [inline s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala:480]:
```suggestion
    val source: Source[ByteString, Any] = Source.single(ByteString(objectValue))
```

**Comment by ennru** [comment]
Some tests fail because of this change as the `/` now gets encoded.
https://travis-ci.com/github/akka/alpakka/jobs/322886269#L605

**Comment by mpdn** [comment]
I corrected HttpRequestSpec to assume urlencoded keys. Alternatively, we could have chosen not to urlencode `/`, but this seems like a simpler solution. There shouldn't be any problem with always urlencoding `/`.

**Comment by ennru** [comment]
Please run `scalafmtAll`.

**Comment by ennru** [comment]
There is another test failure
https://travis-ci.com/github/akka/alpakka/jobs/323013018#L675

**Comment by mpdn** [comment]
That seems odd. I suspect that might actually be a bug in MinIO as it seems like the space is causing problems. Anyway, just removing the space makes the test work at least. Alpakka still supports more characters than before, and AFAIK this should work against S3.

**Comment by ennru** [comment]
Thank you, great fix!

**Pull request merged**